src/qtxdg/xdgmenuapplinkprocessor.cpp

```diff
@@ -192,7 +192,9 @@ void XdgMenuApplinkProcessor::findDesktopFiles(const QString& dirName, const QSt
 
     for (const QFileInfo &file : files)
     {
-        XdgDesktopFile *f = new XdgDesktopFile;
+        mDesktopFiles.push_back(std::make_unique<XdgDesktopFile>());
```

Comment:
Why are you adding the "storage vector" into `XdgMenuApplinkProcessor`... I would say we just need to make the `XdgMenuAppFileInfo` owner of the used `XdgDesktopFile`. Somehting like:
```diff
diff --git a/src/qtxdg/xdgmenuapplinkprocessor.cpp b/src/qtxdg/xdgmenuapplinkprocessor.cpp
index 24646b5..4e078ff 100644
--- a/src/qtxdg/xdgmenuapplinkprocessor.cpp
+++ b/src/qtxdg/xdgmenuapplinkprocessor.cpp
@@ -192,9 +192,9 @@ void XdgMenuApplinkProcessor::findDesktopFiles(const QString& dirName, const QSt
 
     for (const QFileInfo &file : files)
     {
-        XdgDesktopFile *f = new XdgDesktopFile;
+        auto f = std::make_unique<XdgDesktopFile>();
         if (f->load(file.canonicalFilePath()) && f->isValid())
-            mAppFileInfoHash.insert(prefix + file.fileName(), new XdgMenuAppFileInfo(f, prefix + file.fileName(), this));
+            mAppFileInfoHash.insert(prefix + file.fileName(), new XdgMenuAppFileInfo(std::move(f), prefix + file.fileName(), this));
     }
 
 
diff --git a/src/qtxdg/xdgmenuapplinkprocessor.h b/src/qtxdg/xdgmenuapplinkprocessor.h
index bc7ba1b..1f2e765 100644
--- a/src/qtxdg/xdgmenuapplinkprocessor.h
+++ b/src/qtxdg/xdgmenuapplinkprocessor.h
@@ -34,6 +34,7 @@
 #include <QLinkedList>
 #include <QString>
 #include <QHash>
+#include <memory>
 
 class XdgMenu;
 class XdgMenuAppFileInfo;
@@ -80,20 +81,20 @@ class XdgMenuAppFileInfo: public QObject
 {
     Q_OBJECT
 public:
-    explicit XdgMenuAppFileInfo(XdgDesktopFile* desktopFile, const QString& id,  QObject *parent)
+    explicit XdgMenuAppFileInfo(std::unique_ptr<XdgDesktopFile> desktopFile, const QString& id,  QObject *parent)
         : QObject(parent)
+        , mDesktopFile{std::move(desktopFile)}
     {
-        mDesktopFile = desktopFile;
         mAllocated = false;
         mId = id;
     }
 
-    XdgDesktopFile* desktopFile() const { return mDesktopFile; }
+    XdgDesktopFile* desktopFile() const { return mDesktopFile.get(); }
     bool allocated() const { return mAllocated; }
     void setAllocated(bool value) { mAllocated = value; }
     QString id() const { return mId; }
 private:
-    XdgDesktopFile* mDesktopFile;
+    std::unique_ptr<XdgDesktopFile> mDesktopFile;
     bool mAllocated;
     QString mId;
 };
```

Reply:
> I would say we just need to make the XdgMenuAppFileInfo owner of the used XdgDesktopFile

Yeah, that was my first approach too, but I reconsider because semantics seem odd to me that XdgMenuAppFileInfo (auxiliary information of XdgDesktopFile) owns XdgDesktopFile
and preserving semantics is always more important than negligible overhead.

Reply:
> semantics seem odd to me that XdgMenuAppFileInfo (auxiliary information of XdgDesktopFile) owns XdgDesktopFile

Why? Sorry for stupid questions... but I don't see this oddity. I would even remove the `QObject` inheritance ... and inherit directly `XdgDesktopFile`.